### PR TITLE
ManagedSeed: Do not add the Shoot cloud provider credentials into the Seed kubeconfig Secret

### PR DESCRIPTION
**Pull request by ialidzhikov**
/area documentation
/area control-plane
/kind cleanup

I am not aware of cloud provider credentials that should be passed with Seed `spec.secretRef` field. The backup cloud provider credentials are passed via `spec.backup.secretRef`. To me it seems like the doc string is outdated. 

**Release note**:

```other operator
The ManagedSeed controller does no longer add the Shoot cloud provider credentials into the Seed kubeconfig Secret (Seed `.spec.secretRef`).
```

### CODE REVIEW

**Comment by stoyanr** [comment]
@ialidzhikov I believe you are right, just wanted to point out that our seed secrets right now do contain provider credentials, due to https://github.com/gardener/gardener/blob/6095397d3bc75852bf4ad5b86ea5c9dfc9080a20/pkg/gardenlet/controller/managedseed/managedseed_actuator.go#L492

Shouldn't we try to correct this issue first and then update the documentation, to make sure we haven't missed anything? Or perhaps correct them together (removing the above line and at the same time updating the documentation)?

The above line was copied over from the old "seed registration controller" to the new `ManagedSeed` controller. I guess it was obsolete back then, but it didn't cross my mind to double-check this.

@rfranzke Do you know any historical reason why it was like this in the first place?

**Comment by ialidzhikov** [comment]
@stoyanr , actually this is a very nice catch. I agree that we should first tackle the code that is most probably obsolete. Let me try to check whether we can safely remove this handling.
/hold

**Comment by rfranzke** [comment]
I think we didn't have a dedicated `.spec.backup` section in the `Seed` in earlier Gardener versions, so they backup credentials were taken from the secret referenced in `.spec.secretRef`.

File: pkg/gardenlet/controller/managedseed/managedseed_actuator.go [hunk 1]
```diff
@@ -501,7 +497,9 @@ func (a *actuator) createOrUpdateSeedSecrets(ctx context.Context, spec *gardenco
 				*metav1.NewControllerRef(managedSeed, seedmanagementv1alpha1.SchemeGroupVersion.WithKind("ManagedSeed")),
 			}
 			secret.Type = corev1.SecretTypeOpaque
-			secret.Data = data
+			secret.Data = map[string][]byte{
+				kubernetes.KubeConfig: shootKubeconfigSecret.Data[kubernetes.KubeConfig],
+			}
```
Comment (stoyanr) [inline pkg/gardenlet/controller/managedseed/managedseed_actuator.go:502]: Unit tests should be updated as well after this change.

Comment (ialidzhikov) [reply pkg/gardenlet/controller/managedseed/managedseed_actuator.go:502]: Sorry, my bad 🤦‍♂️ .

**Review by stoyanr — APPROVED** [review]
/lgtm

**Review by rfranzke — APPROVED** [review]

**Pull request merged**